cas/src/main/java/org/springframework/security/cas/web/CasAuthenticationFilter.java

```diff
@@ -238,8 +241,14 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
 		}
 		String serviceTicket = obtainArtifact(request);
 		if (serviceTicket == null) {
-			this.logger.debug("Failed to obtain an artifact (cas ticket)");
-			serviceTicket = "";
+			if (this.gateway) {
+				this.logger.debug("Gateway request with no CAS ticket");
+				return new CasGatewayNoServiceTicketAuthenticationToken();
```

Comment:
Am I missing something or this will not be passed to the `CasAuthenticationProvider`?

Reply:
You were right, this was never passed to the `CasAuthenticationProvider`, but this isn't necessary.